src/providers/pkcs11_provider/utils.rs

```diff
@@ -18,16 +18,21 @@ use parsec_interface::requests::ResponseStatus;
 use pkcs11::errors::Error;
 use pkcs11::types::*;
 
+/// Convert the PKCS 11 library specific error values to ResponseStatus values that are returned on
+/// the wire protocol
+///
+/// Most of them are PsaErrorCommunicationFailure as, in the general case, the calls to the PKCS 11
+/// library should never fail. Parsec is totally in control of the input it passes to the library,
+/// if an error happens in the PKCS 11 library, it means that it was badly used by Parsec and hence
+/// the PsaErrorCommunicationFailure error.
```

Comment:
I think that's an overstatement. Sure, some errors would be due to Parsec and how it handles data, but there's a large stack underneath us that could fail in weird ways - other code can have bugs too, not just ours; I think we saw that quite clearly for the TPM stack.

I'm not against this change, but this doc here make it sound as if it's impossible for errors to be returned if we just did everything correctly, and I don't think that's the case. It's probably better if clients don't see the bulk of these errors anyway :) a bit like how servers just return `InternalServerError` as a catch-all for a failure on their part.

Reply:
Will moderate this statement to be more humble.